src/app/components/client/Chart.tsx

```diff
@@ -201,10 +202,30 @@ export const DoughnutChart = (props: Props) => {
   };
 
   const promptContent = getPromptContent();
+  const [centerValueLoaded, setCenterValueLoaded] = useState(false);
+
+  const nrText = document.querySelector(`.${styles.headingNr}`);
+  const labelText = document.querySelector(`.${styles.headingLabel}`);
+
+  // Ignoring this, otherwise tests will complain:
+  // > Warning: A suspended resource finished loading inside a test, but the
+  // > event was not wrapped in act(...).
+  // > When testing, code that resolves suspended data should be wrapped into
+  // > act(...)
+  /* c8 ignore start */
```

Comment:
OK, so I think I figured out why the center value isn't available on page load: it is rendered by a `getFragment`, which replaces `<tags>` in a string with actual HTML tags. However, to do this safely (i.e. to avoid code injection from variables), it uses browser APIs to instantiate DOM elements for those actual tags - and those APIs aren't available during pre-rendering. This is why [`getFragment` only renders on the client-side](https://github.com/mozilla/blurts-server/blob/e5eab8590299ca554f26c80c6f6fe6bad3ed30c7/src/app/hooks/l10n.ts#L20-L22).

So the loading animation while we await the client-side render sounds like a good idea. However, it should be sufficient to use the `useHasRenderedClientSide` hook to determine whether to show it, rather than have the effect dependent on `props.data`, `nrText` and `labelText`. Also, we can add a comment describing the above to explain why we have it :)

And then with https://github.com/mozilla/blurts-server/pull/4551, you should be able to avoid the test from erroring by adding a `jest.mock` call for the `useHasRenderedClientSide` hook.